src/client/pages/forum/forms/PostForm.jsx

```diff
@@ -1,88 +1,99 @@
-import Tag from "../Tag";
-import { useState } from "react";
-import Category from "../Category";
-import {FORUM_URL, LOCAL_STORAGE} from '../../../config';
+import { useState } from 'react';
+
+import Category from '../Category';
+import Tag from '../Tag';
+
+import { FETCH_METHOD, FORUM_URL, LOCAL_STORAGE } from '../../../config';
 
 const PostForm = (props) => {
-	const { posts, setPosts } = props;
-	const [tags, setTags] = useState([]);
-	const [categories, setCategories] = useState([]);
-	const [newPost, setNewPost] = useState({
-		title: "",
-		content: "",
-	});
-
-	const changeHandler = (e) => {
-		const name = e.target.name;
-		const value = e.target.value;
-		setNewPost({ ...newPost, [name]: value });
-
-		if (
-			name === "Gaming" ||
-			name === "Fan-Fiction" ||
-			name === "Cosplay" ||
-			name === "Manga" ||
-			name === "Tv/Film"
-		) {
-			setCategories([...categories, name]);
+    const { posts, setPosts } = props;
+
+	const initialPost = {
+		title: '',
+        content: '',
+	}
+
+    const [tags, setTags] = useState([]);
+    const [categories, setCategories] = useState([]);
+    const [newPost, setNewPost] = useState(initialPost);
+
+    const changeHandler = (event) => {
+        const name = event.target.name;
+        const value = event.target.value;
+
+        setNewPost({ ...newPost, [name]: value });
+
+        if (
+            name === 'Gaming' ||
+            name === 'Fan-Fiction' ||
+            name === 'Cosplay' ||
+            name === 'Manga' ||
+            name === 'Tv/Film'
+        ) {
```

Comment:
I can feel the pain inside you for this. Take a look at `Array.prototype.includes` for an idea about how to make this less gross: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes